lxd/patches.go

```diff
@@ -952,40 +1007,249 @@ func upgradeFromStorageTypeZfs(name string, d *Daemon, defaultPoolName string, d
 		// automatically mounted.
 		output, err := exec.Command("zfs", "set", "mountpoint=none", imageDataset).CombinedOutput()
 		if err != nil {
-			return fmt.Errorf("Failed to set new ZFS mountpoint: %s.", output)
+			shared.LogWarnf("Failed to set new ZFS mountpoint: %s.", output)
+		}
+	}
+
+	var finalErr error
+	if len(failedUpgradeEntities) > 0 {
+		finalErr = fmt.Errorf(strings.Join(failedUpgradeEntities, "\n"))
+	}
+
+	return finalErr
+}
+
+func updatePoolPropertyForAllObjects(d *Daemon, poolName string, allcontainers []string) error {
+	// The new storage api enforces that the default storage pool on which
+	// containers are created is set in the default profile. If it isn't
+	// set, then LXD will refuse to create a container until either an
+	// appropriate device including a pool is added to the default profile
+	// or the user explicitly passes the pool the container's storage volume
+	// is supposed to be created on.
+	profiles, err := dbProfiles(d.db)
+	if err == nil {
+		for _, pName := range profiles {
+			pID, p, err := dbProfileGet(d.db, pName)
+			if err != nil {
+				shared.LogErrorf("Could not query database: %s.", err)
+				return err
+			}
+
+			// Check for a root disk device entry
+			k, _ := containerGetRootDiskDevice(p.Devices)
+			if k != "" {
+				// If found, set the pool property
+				p.Devices[k]["pool"] = poolName
+			} else if k == "" && pName == "default" {
+				// The default profile should have a valid root
+				// disk device entry.
+				rootDev := map[string]string{}
+				rootDev["type"] = "disk"
+				rootDev["path"] = "/"
+				rootDev["pool"] = poolName
+				if p.Devices == nil {
+					p.Devices = map[string]map[string]string{}
+				}
+
+				// Make sure that we do not overwrite a device the user
+				// is currently using under the name "root".
+				rootDevName := "root"
+				for i := 0; i < 100; i++ {
+					if p.Devices[rootDevName] == nil {
+						break
+					}
+					rootDevName = fmt.Sprintf("root%d", i)
+					continue
+				}
+				p.Devices["root"] = rootDev
+			}
+
+			// This is nasty, but we need to clear the profiles config and
+			// devices in order to add the new root device including the
+			// newly added storage pool.
+			tx, err := dbBegin(d.db)
+			if err != nil {
+				return err
+			}
+
+			err = dbProfileConfigClear(tx, pID)
+			if err != nil {
+				shared.LogErrorf("Failed to clear old profile configuration: %s.", err)
+				tx.Rollback()
+				continue
+			}
+
+			err = dbProfileConfigAdd(tx, pID, p.Config)
+			if err != nil {
+				shared.LogErrorf("Failed to add new profile configuration: %s.", err)
+				tx.Rollback()
+				continue
+			}
+
+			err = dbDevicesAdd(tx, "profile", pID, p.Devices)
+			if err != nil {
+				shared.LogErrorf("Failed to add new profile profile root disk device: %s.", err)
+				tx.Rollback()
+				continue
+			}
+
+			err = tx.Commit()
+			if err != nil {
+				shared.LogErrorf("Failed to commit database transaction: %s.", err)
+				tx.Rollback()
+				continue
+			}
```

Comment:
Same comment as before, none of those errors tell me what profile failed to update so how am I supposed to fix things by hand?